crates/node/builder/src/launch/engine_task.rs

```diff
@@ -0,0 +1,211 @@
+        event_sender: EventSender<BeaconConsensusEngineEvent>,
+        initial_backfill_target: Option<B256>,
+        payload_validation_enabled: bool,
+    ) -> Self {
+        Self {
+            chain_spec,
+            network_handle,
+            built_payload_stream,
+            service,
+            event_sender,
+            initial_backfill_target,
+            payload_validation_enabled,
+            new_payload_responses: FuturesUnordered::default(),
+        }
+    }
+
+    fn on_chain_event(&self, event: ChainEvent<BeaconConsensusEngineEvent>) -> eyre::Result<()> {
+        debug!(target: "reth::cli", "Event: {event}");
+        match event {
+            ChainEvent::BackfillSyncFinished => {
+                self.network_handle.update_sync_state(SyncState::Idle);
+            }
+            ChainEvent::BackfillSyncStarted => {
+                self.network_handle.update_sync_state(SyncState::Syncing);
+            }
+            ChainEvent::FatalError => {
+                error!(target: "reth::cli", "Fatal error in consensus engine");
+                return Err(eyre::eyre!("Fatal error in consensus engine"));
+            }
+            ChainEvent::Handler(ev) => {
+                if let Some(head) = ev.canonical_header() {
+                    let head_block = Head {
+                        number: head.number,
+                        hash: head.hash(),
+                        difficulty: head.difficulty,
+                        timestamp: head.timestamp,
+                        total_difficulty: self
+                            .chain_spec
+                            .final_paris_total_difficulty(head.number)
+                            .unwrap_or_default(),
+                    };
+                    self.network_handle.update_status(head_block);
+                }
+                self.event_sender.notify(ev);
+            }
+        }
+        Ok(())
+    }
+
+    fn on_built_executed_block(&mut self, executed_block: ExecutedBlock) {
+        let request = if self.payload_validation_enabled {
+            debug!(target: "reth::cli", block=?executed_block.block().num_hash(), "inserting built payload with validation");
+            let (tx, rx) = oneshot::channel();
+            self.new_payload_responses.push(rx);
+
+            let message = BeaconEngineMessage::NewPayload {
+                payload: block_to_payload(executed_block.block.as_ref().clone()),
+                cancun_fields: executed_block.block.parent_beacon_block_root.map(
+                    |parent_beacon_block_root| CancunPayloadFields {
+                        parent_beacon_block_root,
+                        versioned_hashes: executed_block
+                            .block
+                            .blob_versioned_hashes_iter()
+                            .copied()
+                            .collect(),
+                    },
+                ),
+                tx,
+            };
+            EngineApiRequest::Beacon(message)
```

Comment:
this is the only thing that changed right? 
we add it as a different request variant and log the response

imo we could make this a lot simpler if we do this with the main version and simply spawn a new task that awaits this response and logs it, because I really prefer the loop select! for this